### ISSUE TITLE
Query loop block: fix block locking flow for consistency and to prevent breaking out of locking

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#39908]
## What problem does this address?

For some of the more advanced blocks, there are what I'll call setup states where you need to take a few steps in order to configure the block in question. Three examples are the navigation block, query loop block, and template part block. Currently, if you lock these blocks during the initial setup stage, different outcomes occur:

**Query Loop: doesn't lock**

https://user-images.githubusercontent.com/26996883/160897305-428ad4ed-bcff-40fc-8faa-bf7988b08c37.mov

**Navigation block: does lock**

https://user-images.githubusercontent.com/26996883/160897474-e3e2135e-61ba-48a5-81fd-cacbe8995518.mov

**Template Part block: does lock**

https://user-images.githubusercontent.com/26996883/160897792-87e479e7-8c6e-477b-a0d8-7e8dcb370983.mov

This testing was all done with Gutenberg 12.9 and WordPress 5.9.2 using TT2. 

## What is your proposed solution?

I'm not quite sure what's expected here but I think we need to audit any additional blocks that have this setup state in case someone tries to look early on and have consistency in the outcome. In this case, the outcome could be as simple as updating the behavior of the Query Loop block to be consistent with the other two.

cc @Mamaduka

**Comment by Mamaduka** [comment]
Thanks for the ping, @annezazu.

Navigation and Template Parts setup works differently from Query block. Those are still the same blocks, but the reference template or navigation you just chose.

```html
<!-- wp:template-part {"slug":"untitled-template-part","theme":"twentytwentytwo"} /-->

<!-- wp:navigation {"ref":72} /-->
```

The Query block, on the other hand, replaces the whole block with the chosen pattern from the setup.

Currently, I can think of two options:

1. Disable locking when Query block is in setup state.
2. Pass the lock state to the replacement Query pattern.

Cc @ntsekouras.

**Comment by ntsekouras** [comment]
>The Query block, on the other hand, replaces the whole block with the chosen pattern from the setup.
>Currently, I can think of two options:
>1. Disable locking when Query block is in setup state.
>2. Pass the lock state to the replacement Query pattern.

We should find a way to implement the first option - and just for reference this is when there are no innerBlocks in Query Loop.

We can't do something about the second option, as we replace a single Query Loop block with a pattern. What this means is there could be any blocks, from a wrapper Group block to a paragraph and then a Query Loop(or even multiple ones), so we can't even rely on the first replaced block etc..

**Comment by Mamaduka** [comment]
@ntsekouras do you think it makes sense to hide the whole `BlockSettingsMenuControls` when Query Loop is in the setup state?

![CleanShot 2022-04-03 at 18 48 54](https://user-images.githubusercontent.com/240569/161433706-d3c27444-a6f8-41c1-8921-bfd5d2db683a.png)

**Comment by ntsekouras** [comment]
>@ntsekouras do you think it makes sense to hide the whole BlockSettingsMenuControls when Query Loop is in the setup state?

For me it makes sense to hide the highlighted part of your screenshot, yes.

**Comment by annezazu** [comment]
Testing this with GB 22.3 and WP 6.9 and, due to the current setup with the Query Loop block, I can actually break out of the block locking with the "change design" option available when the block is still locked:

https://github.com/user-attachments/assets/6ec64fc0-c7c0-49de-a2de-148e5864f26b

Going to update this issue to be about the query loop block only at this point as we need two things to happen:
- Remove the ability to lock the query loop block during setup
- Remove the "change design" option when the block is locked as being able to switch designs removes the locking

**Comment by ntsekouras** [comment]
I opened a [PR](https://github.com/WordPress/gutenberg/pull/74160) for `Remove the "change design" option when the block is locked as being able to switch designs removes the locking`.

For `Remove the ability to lock the query loop block during setup`, I couldn't figure out a good way to remove the `lock` menu item for our specific case. I thought about allowing a block to set the locking state itself, but that would be a quite big change for a small benefit. I also didn't want to hardcode exceptions. Any suggestions are welcome.

**Comment by Mamaduka** [comment]
I think the second part ("hiding setting for setup state") will be a bit hard to resolve. This state is unique to each block and is difficult to generalize at the Block Editor package level.

I would say we close this as "maybe later".

**Comment by annezazu** [comment]
I agree. It feels a bit of an edge case as well whereas the first issue truly breaks out of the locked mode. Closing!
